client/src/main/scala/org/http4s/client/PoolManager.scala

```diff
@@ -147,6 +162,9 @@ private final class PoolManager[F[_], A <: Connection[F]](
                 logger.debug(s"Active connection not found. Creating new one. $stats")
                 createConnection(key, callback)
 
+              case None if maxConnectionsPerRequestKey(key) <= 0 =>
```

Comment:
This is a new enhancement, but it prevents a hang in the spec, so bolted on to this.